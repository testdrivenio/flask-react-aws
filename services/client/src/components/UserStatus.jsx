import React, { Component } from "react";
import axios from "axios";
import PropTypes from "prop-types";
import { Redirect } from "react-router-dom";

class UserStatus extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      username: ""
    };
  }
  componentDidMount() {
    this.getUserStatus();
  }
  getUserStatus(event) {
    const options = {
      url: `${process.env.REACT_APP_USERS_SERVICE_URL}/auth/status`,
      method: "get",
      headers: {
        "Content-Type": "application/json",
        Authorization: `Bearer ${this.props.accessToken}`
      }
    };
    return axios(options)
      .then(res => {
        this.setState({
          email: res.data.email,
          username: res.data.username
        });
      })
      .catch(error => {
        console.log(error);
      });
  }
  render() {
    if (!this.props.isAuthenticated()) {
      return <Redirect to="/login" />;
    }
    return (
      <div>
        <ul>
          <li>
            <strong>Email:</strong>&nbsp;
            <span data-testid="user-email">{this.state.email}</span>
          </li>
          <li>
            <strong>Username:</strong>&nbsp;
            <span data-testid="user-username">{this.state.username}</span>
          </li>
        </ul>
      </div>
    );
  }
}

UserStatus.propTypes = {
  accessToken: PropTypes.string,
  isAuthenticated: PropTypes.func.isRequired
};

export default UserStatus;
