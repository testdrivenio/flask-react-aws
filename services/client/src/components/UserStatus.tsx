import React, { useState, useEffect } from "react";
import axios from "axios";
import { Box } from "@chakra-ui/react";
import { Navigate } from "react-router-dom";

interface UserStatusProps {
  accessToken: string;
  isAuthenticated: () => boolean;
}

const UserStatus: React.FC<UserStatusProps> = ({
  accessToken,
  isAuthenticated,
}) => {
  const [email, setEmail] = useState("");
  const [username, setUsername] = useState("");

  useEffect(() => {
    const getUserStatus = async () => {
      try {
        const options = {
          url: `${import.meta.env.VITE_API_SERVICE_URL}/auth/status`,
          method: "get",
          headers: {
            "Content-Type": "application/json",
            Authorization: `Bearer ${accessToken}`,
          },
        };
        const res = await axios(options);
        setEmail(res.data.email);
        setUsername(res.data.username);
      } catch (error) {
        console.log(error);
      }
    };
    getUserStatus();
  }, [accessToken]);

  if (!isAuthenticated()) {
    return <Navigate to="/login" replace />;
  }

  return (
    <Box p={4} maxW="1200px" mx="auto" mt={12}>
      <div>
        <p data-testid="user-email">{email}</p>
        <p data-testid="user-username">{username}</p>
      </div>
    </Box>
  );
};

export default UserStatus;
