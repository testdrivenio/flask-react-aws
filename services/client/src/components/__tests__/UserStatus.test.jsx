import React from 'react';
import { cleanup, wait } from '@testing-library/react';
import axios from 'axios';

import UserStatus from '../UserStatus';

afterEach(cleanup);

jest.mock("axios");

axios.mockImplementation(() =>
  Promise.resolve({
    data: { email: 'test@test.com', username: 'test' }
  })
);

const props = {
  isAuthenticated: () => { return true },
}

it('renders properly when authenticated', async() => {

  const { container, findByTestId } = renderWithRouter(<UserStatus {...props} />);
  await wait(() => {
    expect(axios).toHaveBeenCalledTimes(1);
  });
  expect((await findByTestId('user-email')).innerHTML).toBe('test@test.com');
  expect((await findByTestId('user-username')).innerHTML).toBe('test');
});

it("renders", async() => {
  const { asFragment } = renderWithRouter(<UserStatus {...props}  />);  // updated
  await wait(() => {
    expect(axios).toHaveBeenCalled();
  });
  expect(asFragment()).toMatchSnapshot();
});
