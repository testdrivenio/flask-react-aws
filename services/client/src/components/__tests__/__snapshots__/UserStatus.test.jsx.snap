// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <ul>
      <li>
        <strong>
          Email:
        </strong>
         
        <span
          data-testid="user-email"
        >
          test@test.com
        </span>
      </li>
      <li>
        <strong>
          Username:
        </strong>
         
        <span
          data-testid="user-username"
        >
          test
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
